gson/src/main/java/com/google/gson/JsonArray.java

```diff
@@ -151,6 +151,28 @@ public JsonElement remove(int index) {
   public boolean contains(JsonElement element) {
     return elements.contains(element);
   }
+  
+  /**
+   * Overloaded convenience method which searches a string in JsonArray and returns true if it exists
+   * Assumes that the JsonArray contains string elements
+   * @return true if this array contains the specified element.
+   * @param element whose presence in this array is to be tested
+   */
+  public boolean contains(String element) {
+     if(element==null || elements.size()==0 ) return false;
+     return contains(new JsonPrimitive(element));     
+  }
+  
+  /**
+   * Overloaded convenience method which searches a Integer in JsonArray and returns true if it exists
+   * Assumes that the JsonArray contains integer elements
+   * @return true if this array contains the specified element.
+   * @param element whose presence in this array is to be tested
+   */
+  public boolean contains(Integer element) {
```

Comment:
Json array can contain only integer, string , json objects. Most of us use contains for checking String,Integer. Hence only 2 primitives.